kernel: mmu: fix compilation warnings when memory size is 0

ycsin opened the pull request:
When these memory sizes are defined to be zero, the assertions test will be comparing an unsigned int against zero which result in compilation warning, and will be raised to error in Twister.

Fix them with more conditional compilations, see also #63080

fixes #64633

andyross reviewed (commented):
Putting #if's around single arguments in a macro is sort of messy, especially in code like this that's already heavily #if'd.  Can't we refactor this to be readable, somehow?  They're assertions, they're supposed to be a clear expression of some kind of truth, right?  So when they fail we know how to fix it?

ycsin commented:
> Putting #if's around single arguments in a macro is sort of messy, especially in code like this that's already heavily #if'd. Can't we refactor this to be readable, somehow? They're assertions, they're supposed to be a clear expression of some kind of truth, right? So when they fail we know how to fix it?

True, in less than 10 minutes after creating this PR, I doubt if I have the full picture to produce a legit fix, therefore I pulled this back to draft

ycsin commented on include/zephyr/sys/mem_manage.h:
```diff
@@ -194,17 +194,21 @@ static inline uintptr_t z_mem_phys_addr(void *virt)
 #if CONFIG_KERNEL_VM_BASE != 0
 		 (addr >= CONFIG_KERNEL_VM_BASE) &&
 #endif
-		 (addr < (CONFIG_KERNEL_VM_BASE +
-			  (CONFIG_KERNEL_VM_SIZE))),
```
When both `CONFIG_KERNEL_VM_BASE` and `CONFIG_KERNEL_VM_SIZE` are 0, `addr` will never be lesser than 0 and therefore it would be essentially:

```c
__ASSERT(false, ...);

nashif commented:
why is this a hotfix? Is the issue breaking CI somewhere?

ycsin commented:
> why is this a hotfix? Is the issue breaking CI somewhere?

Yes - https://github.com/zephyrproject-rtos/zephyr/actions/runs/6705912230/job/18221509407?pr=63172

aescolar commented:
> > why is this a hotfix? Is the issue breaking CI somewhere?
> 
> Yes - https://github.com/zephyrproject-rtos/zephyr/actions/runs/6705912230/job/18221509407?pr=63172

A hotfix implies there is an issue in the plain main branch itself blocking development. That is, that tests do not pass on main itself.
It seems that failure you are pointing at is in your branch due to your branch changes(?).

=> Removing the hotfix label.

ycsin commented:
> It seems that failure you are pointing at is in your branch due to your branch changes(?).

I think you are right, the warning was due to the inclusion of an additional `#include <zephyr/device.h>` in `sw_isr_table.h`

ycsin committed:
kernel: mmu: fix compilation warnings when memory address and size are 0

When both the memory base address & its size are zero, the
assertions test will be comparing an unsigned int against zero
which result in compilation warning, and will be raised to
error in Twister

Fix them with more conditional compilations.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>
